nusacrowd/nusa_datasets/su_emot/su_emot.py

```diff
@@ -0,0 +1,131 @@
+validation, the authors consulted a Sundanese language teacher for expert validation.
+"""
+_HOMEPAGE = "https://github.com/virgantara/sundanese-twitter-dataset"
+
+_LICENSE = "UNKNOWN"
+
+_URLS = {
+	"datasets": "https://raw.githubusercontent.com/virgantara/sundanese-twitter-dataset/master/newdataset.csv"
+}
+
+_SUPPORTED_TASKS = [Tasks.EMOTION_CLASSIFICATION]
+
+
+_SOURCE_VERSION = "1.0.0"
+
+_NUSANTARA_VERSION = "1.0.0"
+
+
+class suEmot(datasets.GeneratorBasedBuilder):
+    """This is a dataset for emotion classification of Sundanese text. The dataset is gathered from Twitter API between January and March 2019 with 2518 tweets in total."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    NUSANTARA_VERSION = datasets.Version(_NUSANTARA_VERSION)
+
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="su_emot_source",
+            version=SOURCE_VERSION,
+            description="Sundanese Twitter Dataset for Emotion source schema",
+            schema="source",
+            subset_id="su_emot",
+        ),
+        NusantaraConfig(
+            name="su_emot_nusantara_text",
+            version=NUSANTARA_VERSION,
+            description="Sundanese Twitter Dataset for Emotion Nusantara schema",
+            schema="nusantara_text",
+            subset_id="su_emot",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "su_emot_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features({
+                "index": datasets.Value("string"),
+                "data": datasets.Value("string"), 
+                "label": datasets.Value("string")})
+
+        # For example nusantara_kb, nusantara_t2t
+        elif self.config.schema == "nusantara_text":
+            features = schemas.text_features(["anger", "joy", "fear", "sadness"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        urls = _URLS
+        data_dir = Path(dl_manager.download_and_extract(urls['datasets']))
+        data_files = {"train":data_dir}
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_files['train'],
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+
+        df = pd.read_csv(filepath, sep=",", header="infer").reset_index()
+        df.columns = ["index","label", "data"]
+
+        if self.config.schema == "source":
+            for row in df.itertuples():
+                ex = {"index": str(row.index), "data": row.data, "label": row.label}
```

Comment:
```suggestion
                ex = {"index": str(row.index+1), "data": row.data, "label": row.label}
```

For `index`, I think will be better if the number if from `1` to `n`